packages/squiggle-lang/src/value/valueTags.ts

```diff
@@ -1,23 +1,23 @@
-import { result } from "../index.js";
 import { ImmutableMap } from "../utility/immutableMap.js";
-import { Err, fmap, mergeMany, Ok } from "../utility/result.js";
-import { Value, vArray, vBool, vDict, VDict, vString } from "./index.js";
+import { Err, fmap, mergeMany, Ok, result } from "../utility/result.js";
+import { Value } from "./index.js";
+import { type VBool } from "./VBool.js";
+import { type VDict } from "./VDict.js";
+import { type VString } from "./VString.js";
 
-export type ExportData = {
-  sourceId: string;
-  path: string[];
-};
+// Note: this file can't call any `vType` constructors; it would cause a circular dependency because of `BaseValue` -> `ValueTags`.
 
+// Some of these tags are value-type-specific, but we store everything in a single Dict, for now.
 export type ValueTagsType = {
-  name?: string;
-  doc?: string;
+  name?: VString;
+  doc?: VString;
   showAs?: Value;
-  numberFormat?: string;
-  dateFormat?: string;
-  hidden?: boolean;
-  notebook?: boolean;
-  exportData?: ExportData;
-  startOpenState?: "open" | "closed";
+  numberFormat?: VString; // can be set on numbers, dists and durations
+  dateFormat?: VString; // can be set on dates
+  hidden?: VBool;
+  notebook?: VBool; // can be set on arrays
+  exportData?: VDict; // should be { sourceId: String, path: List(String) }
```

Comment:
maybe when we eventually have Structs, we could use them here. Not sure though.

Reply:
Yep, I mentioned this idea in description